Drop SLE Micro version references from code

davidcassany opened the pull request:
These are the changes required in this repository for the builds I prepared in [here](https://build.opensuse.org/project/show/home:dcassany:ElementalNewDev). 

Note this changes image and charts URLs, so this is likely to be a breaking change from CI perspective.

Part of #815

davidcassany committed:
Drop SLE Micro version references from code

Signed-off-by: David Cassany <dcassany@suse.com>

kkaempf commented on .obs/dockerfile/teal-iso/Dockerfile:
```diff
@@ -1,12 +1,14 @@
-#!BuildTag: rancher/elemental-teal-iso/5.3:latest
-#!BuildTag: rancher/elemental-teal-iso/5.3:%VERSION%
-#!BuildTag: rancher/elemental-teal-iso/5.3:%VERSION%-%RELEASE%
+#!BuildTag: rancher/elemental-teal-iso:latest
```
Hmm, this might fail while we build two versions (like `5.3` and `5.4`) in parallel.

We should probably just drop the `:latest` tag ?!

davidcassany replied:
In OBS builds this will end up being something like:
```
registry.org/home/cassany/elementalnewdev/containers/rancher/elemental-teal-iso:latest
registry.org/home/cassany/elementalnewdev/containers_next/rancher/elemental-teal-iso:latest
```

So there will not be an overlap as long as they build in different repositories. For IBS builds under registry.suse.com I see your point. If we follow other SUSE images approach we could keep it our latest code base stream build and remove it on old ones. Using `suse/sle15:latest` image as an example this shifted from SP4 to SP5 as soon as the new one was released. I agree this is quite annoying from maintenance point of view as this forces us to update the old build in any new release 🤔 also feels like this can be error prone if done manually.

kkaempf commented on .obs/dockerfile/teal-iso/Dockerfile:
```diff
@@ -1,12 +1,14 @@
-FROM rancher/elemental-builder-image/5.3:latest AS builder
+ARG SLE_VERSION
+
+FROM rancher/elemental-teal:latest AS os
```
Same here ...

davidcassany committed:
Remove latest tags from teal-iso and teal images

Signed-off-by: David Cassany <dcassany@suse.com>

davidcassany committed:
Readd SLE Micro version within image repository

Add again SLEMicro version within the repository to be sure we can still
make use of the 'latest' tag for each base distro version. Now it will
be dynamically computed from project configuration at the expense of
increasing _service file complexity in OBS.

Signed-off-by: David Cassany <dcassany@suse.com>

kkaempf reviewed (approved):
Looks much better now 😆
